src/services/gfn.service.ts

```diff
@@ -53,33 +46,23 @@ export interface ResponseError {
 }
 
 class GfnService {
-  private readonly APIURL =
-    'https://us-central1-gfn-games.cloudfunctions.net/gfn';
+  /**
+   * Helper function to convert Firestore data into an Observable
+   * @param query
+   */
+  private _getData$(
+    query: firebase.firestore.Query<firebase.firestore.DocumentData>,
+  ): Observable<Game[]> {
+    return fromEventPattern(
+      (handler) => query.onSnapshot(handler),
+      (handler, unsubscribe) => unsubscribe(),
+    ).pipe(map((data: any) => data.docs.map((doc: any) => doc.data() as Game)));
+  }
 
   get games$(): Observable<Game[] | ResponseError> {
-    const url: string = `${this.APIURL}/GetAllGames`;
+    const query = db.collection('gfn');
 
-    return fromFetch(url).pipe(
-      switchMap((response: Response) => {
-        if (!response.ok) {
-          // Server is returning a status requiring the client to try something else.
-          return of({ error: true, message: `Error ${response.status}` });
-        }
-        return response.json();
-      }),
-      // Add source to the game model
-      map((games: Game[]) => {
-        return games.map((game: Game) => {
-          return {
-            ...game,
-            appid: game.steamUrl
-              ? Number(game.steamUrl.split('/').pop())
-              : null,
-            source: 'GFN',
-            free: game.genres.includes('Free To Play'),
-          } as Game;
-        });
-      }),
+    return this._getData$(query).pipe(
```

Comment:
I like how you abstracted this away.

Reply:
Yeah. With AngularFire you just do query.valueChanges() and get an observable returned. I see there is a `vue-firestore` npm package, but this is all we need for now.